kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala

```diff
@@ -134,6 +134,19 @@ class KubernetesApplicationOperation extends ApplicationOperation with Logging {
   }
 }
 
-object KubernetesApplicationOperation {
+object KubernetesApplicationOperation extends Logging {
   val LABEL_KYUUBI_UNIQUE_KEY = "kyuubi-unique-tag"
+
+  def toApplicationState(state: String): ApplicationState = state match {
+    // https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L2396
+    // https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/
+    case "Pending" => PENDING
+    case "Running" => RUNNING
+    case "Succeeded" => FINISHED
+    case "Failed" | "Error" => FAILED
+    case "Unknown" => ApplicationState.UNKNOWN
+    case _ =>
+      warn(s"Not support state: $state, using UNKNOWN")
```

Comment:
Please make the warning message meaningful and illustrative if you are warning people.

> The kubernetes driver pod state: $state is not supported, mark the application state as UNKNOWN.

Reply:
updated